src/FastRouteRouter.php

```diff
@@ -405,11 +440,21 @@ private function injectRoute(Route $route)
         $methods = $route->getAllowedMethods();
 
         if ($methods === Route::HTTP_METHOD_ANY) {
-            $methods = ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'HEAD', 'OPTIONS', 'TRACE'];
+            $methods = self::HTTP_METHODS_STANDARD;
+        }
+
+        // Remove HEAD from allowed methods if it was implicitly registered
+        if (in_array(RequestMethod::METHOD_HEAD, $methods, true) && $route->implicitHead()) {
```

Comment:
There's no reason to remove the method if it's not present in the first place. However, it's a moot point, as auto-adding the HEAD and OPTIONS methods when creating the `Route` instance was causing problems with route duplication in the `Application` instance. As such, this check is no longer needed; we'll check for `implicitHead()` and/or `implicitOptions()` in the dispatcher middleware instead. I've now removed these conditionals.